core/src/main/java/com/linecorp/armeria/server/encoding/EncodingServiceBuilder.java

```diff
@@ -0,0 +1,128 @@
+                                 .anyMatch(contentType::is);
+
+    private static final int DEFAULT_MIN_BYTES_TO_FORCE_CHUNKED_AND_ENCODING = 1024;
+
+    private Predicate<MediaType> encodableContentTypePredicate = defaultEncodableContentTypePredicate;
+
+    private Predicate<RequestHeaders> encodableRequestHeadersPredicate = headers -> true;
+
+    private int minBytesToForceChunkedAndEncoding = DEFAULT_MIN_BYTES_TO_FORCE_CHUNKED_AND_ENCODING;
+
+    EncodingServiceBuilder() {}
+
+    /**
+     * Sets the specified {@link MediaType}s to evaluate whether the content type of the {@link HttpResponse}
+     * is encodable or not. It's encodable when the content type is one of the {@link MediaType}s.
+     */
+    public EncodingServiceBuilder encodableContentTypes(MediaType... contentTypes) {
+        return encodableContentTypes(ImmutableList.copyOf(requireNonNull(contentTypes, "contentTypes")));
+    }
+
+    /**
+     * Sets the specified {@link MediaType}s to evaluate whether the content type of the {@link HttpResponse}
+     * is encodable or not. It's encodable when the content type is one of the {@link MediaType}s.
+     */
+    public EncodingServiceBuilder encodableContentTypes(Iterable<MediaType> contentTypes) {
+        final List<MediaType> snapshot = ImmutableList.copyOf(requireNonNull(contentTypes, "contentTypes"));
+        return encodableContentTypePredicate(mediaType -> snapshot.stream().anyMatch(mediaType::belongsTo));
+    }
+
+    /**
+     * Sets the specified {@link Predicate} to evaluate whether the content type of the {@link HttpResponse} is
+     * encodable or not.
+     */
+    public EncodingServiceBuilder encodableContentTypePredicate(
+            Predicate<MediaType> encodableContentTypePredicate) {
+        requireNonNull(encodableContentTypePredicate, "encodableContentTypePredicate");
+        this.encodableContentTypePredicate = encodableContentTypePredicate;
+        return this;
+    }
+
+    /**
+     * Sets the specified {@link Predicate} to evaluate whether the corresponding {@link HttpResponse} of the
+     * {@link HttpRequest} whose {@link RequestHeaders} is the input of the {@link Predicate}
+     * is encodable or not.
+     */
+    public EncodingServiceBuilder encodableRequestHeadersPredicate(
+            Predicate<RequestHeaders> encodableRequestHeadersPredicate) {
+        requireNonNull(encodableRequestHeadersPredicate, "encodableRequestHeadersPredicate");
+        this.encodableRequestHeadersPredicate = encodableRequestHeadersPredicate;
+        return this;
+    }
+
+    /**
+     * Sets the specified minimum length to force chunked encoding. The {@link HttpResponse} is encoded only
+     * when the content is variable, which means the {@link ResponseHeaders} does not have
+     * {@code "Content-Length"} header, or the length of the content exceeds the specified length.
+     * The default is {@value DEFAULT_MIN_BYTES_TO_FORCE_CHUNKED_AND_ENCODING}.
```

Comment:
`DEFAULT_MIN_BYTES_TO_FORCE_CHUNKED_AND_ENCODING` is a private field.

Reply:
It's OK to be a private field. `@value` will inline the value.

Reply:
Yes it does not have to be. 😉

Reply:
Ah, I didn't know that, I've learned a lot. Thanks!